docs/sphinx/user_guide/using_camp.rst

```diff
@@ -0,0 +1,132 @@
+.. _using_camp-label: 
+
+**********
+Using Camp
+**********
+
+Since Camp is a metaprogramming library that is primarily header-only, it is hard
+to find isolated code examples that show Camp's capability. Instead, this page
+shows several real application examples of how Camp is being used currently.
+
+Camp Used in Umpire
+===================
+
+Umpire is a resource management library that allows the discovery, provision, and management of memory on machines 
+with multiple memory devices like NUMA and GPUs. Umpire is also part of the RAJA Portability Suite.
+Umpire's Operations provide an abstract interface to modifying and moving data between Umpire allocators.
+Camp makes these operations easy to port regardless of the underlying hardware. For example:
+
+.. code-block:: cpp
+
+  camp::resources::EventProxy<camp::resources::Resource> CudaMemsetOperation::apply_async(
+    void* src_ptr, util::AllocationRecord* UMPIRE_UNUSED_ARG(allocation), int value, std::size_t length,
+    camp::resources::Resource& ctx)
+  {
+    auto device = ctx.try_get<camp::resources::Cuda>();
+    if (!device) {
+      UMPIRE_ERROR(resource_error,
+                 fmt::format("Expected resources::Cuda, got resources::{}", platform_to_string(ctx.get_platform())));
+    }
+    auto stream = device->get_stream();
+
+    cudaError_t error = ::cudaMemsetAsync(src_ptr, value, length, stream);
+
+    if (error != cudaSuccess) {
+      UMPIRE_ERROR(
+        runtime_error,
+        fmt::format("cudaMemsetAsync( src_ptr = {}, value = {}, length = {}, stream = {}) failed with error: {}",
+                    src_ptr, value, length, cudaGetErrorString(error), (void*)stream));
+    }
+
+    return camp::resources::EventProxy<camp::resources::Resource>{ctx};
+  }
+
+See the full example `here <https://github.com/LLNL/Umpire/blob/5bf5bc182f1e6ee3f6be1d953b68451d3ddc35f5/src/umpire/op/CudaMemsetOperation.cpp>`_.
+
+.. note::
+
+   The new ``ResourceAwarePool`` feature in Umpire will be using both Camp resources and Camp events to
+   keep track of the state of different allocations within the pool. Without these Camp features, it would
+   be much harder to keep track of the state of the allocations in a portable, hardware-agnostic way.
+
+Camp Used in RAJA
+=================
+
+RAJA is a library of C++ software abstractions, primarily developed at Lawrence Livermore National Laboratory (LLNL), that enables 
```

Comment:
```suggestion
`RAJA <https://github.com/LLNL/RAJA>`_ is a library of C++ software abstractions that enables 
```
The clause about where it is developed will be implicit by linking in the RAJA project home page.